### PR TITLE
refactor: Initialize template variables for LaTeX in builder; date, logofilename and packages 

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Refactoring

### Purpose
- Simplify LaTeX writer

### CODE REVIEW

File: sphinx/templates/latex/latex.tex_t [hunk 1]
```diff
@@ -48,7 +56,9 @@
 \date{<%= date %>}
 \release{<%= release %>}
 \author{<%= author %>}
-\newcommand{\sphinxlogo}{<%= logo %>}
+<%- if logofilename %>
+\newcommand{\sphinxlogo}{\sphinxincludegraphics{<%= logofilename %>}\par}
+<%- endif %>
```
Comment (jfbu) [inline sphinx/templates/latex/latex.tex_t:63]:
```
<%- if logofilename %>
\newcommand{\sphinxlogo}{\sphinxincludegraphics{<%= logofilename %>}\par}
<%- else %>
\newcommand{\sphinxlogo}{\vbox{}}
<%- endif %>
```

**Review by jfbu — CHANGES_REQUESTED** [review]
LGTM with nits

File: sphinx/templates/latex/latex.tex_t (hunk 1 above)
Comment (tk0miya) [reply sphinx/templates/latex/latex.tex_t:63]: oh, thanks!

Comment (jfbu) [reply sphinx/templates/latex/latex.tex_t:63]: @tk0miya it is also needed to remove mention of ``'logo'`` from this

```
* Keys that are set by other options and therefore should not be overridden
     are:

     ``'docclass'``
     ``'classoptions'``
     ``'title'``
     ``'date'``
     ``'release'``
     ``'author'``
     ``'logo'``
     ``'makeindex'``
```

at lines about L2319-2329 in https://github.com/sphinx-doc/sphinx/blob/master/doc/usage/configuration.rst, and also ``'date'``

Comment (tk0miya) [reply sphinx/templates/latex/latex.tex_t:63]: done!

**Review by jfbu — APPROVED** [review]
LGTM!

**Comment by tk0miya** [comment]
Merged. Thank you for reviewing!

**Pull request merged**